codegen/velocity/src/main/java/io/sundr/codegen/processor/TemplateGeneratingProcessor.java

```diff
@@ -106,8 +109,12 @@ public void generateFromStringTemplate(TypeDef model, String[] parameters, Strin
       printSkipping(newModel.getFullyQualifiedName(), "Class already exists.");
       return;
     }
-    generateFromStringTemplate(model, parameters, processingEnv.getFiler().createSourceFile(newModel.getFullyQualifiedName()),
-        content);
+
+    CodeGenerator.newGenerator(TypeDef.class)
+        .withIdentifier(TypeDef::getFullyQualifiedName)
+        .withOutput(new TypeDefAptOutput(processingEnv.getFiler()))
+        .withRenderer(VelocityRenderer.fromTemplate(content, parameters))
+        .generate(newModel);
```

Comment:
Is the function in line 129 still used (`generateFromStringTemplate(T model, String[] parameters, FileObject fileObject, String content)`)?
If not, shouldn't it be marked for deprecation?

Reply:
The class is removed.